Gems/Atom/RHI/Code/Include/Atom/RHI/BufferPool.h

```diff
@@ -23,7 +23,59 @@ namespace AZ::RHI
         AZStd::unordered_map<int, void*> m_data;
     };
 
-    using BufferInitRequest = BufferInitRequestTemplate<Buffer>;
+    struct BufferInitRequest
+    {
+        BufferInitRequest() = default;
+
+        BufferInitRequest(
+            Buffer& buffer,
+            const BufferDescriptor& descriptor,
+            const void* initialData = nullptr,
+            MultiDevice::DeviceMask deviceMask = MultiDevice::AllDevices)
+            : m_buffer{ &buffer }
+            , m_descriptor{ descriptor }
+            , m_initialData{ initialData }
+            , m_deviceMask{ deviceMask }
+        {
+        }
+
+        /// The buffer to initialize. The buffer must be in an uninitialized state.
+        Buffer* m_buffer = nullptr;
+
+        /// The descriptor used to initialize the buffer.
+        BufferDescriptor m_descriptor;
+
+        /// [Optional] Initial data used to initialize the buffer.
+        const void* m_initialData = nullptr;
+
+        /// The device mask used for the buffer.
+        /// Note: Only devices in the mask of the buffer pool will be considered.
+        MultiDevice::DeviceMask m_deviceMask = MultiDevice::AllDevices;
+    };
+
+    struct BufferDeviceMaskRequest
```

Comment:
This structure is essentially a copy same as BufferInitRequest with the exception of BufferDescriptor so why not use this inside BufferInitRequest instead of duplicating code?